src/taipy/core/data/excel.py

```diff
@@ -287,20 +287,29 @@ def _read_as_modin_dataframe(
         except pd.errors.EmptyDataError:
             return modin_pd.DataFrame()
 
+    def __write_excel_with_sheet_name(self, write_excel_fct, *args, **kwargs):
+        sheet_name = self.properties.get(self.__SHEET_NAME_PROPERTY)
+        if sheet_name:
+            sheet_name = sheet_name if isinstance(sheet_name, str) else sheet_name[0]
```

Comment:
This line looks suspicious to me. 

Why do we only write the first one if there are multiple sheet names? 
The number of data frames to write should be the same as the nb of sheet names. No?
Since this method is only called when the data is not a dictionary of data frames, that means it is only used with single-sheet Excel files. This is inconsistent with multiple sheet names. No?
In other words, do we want to raise an error when writing data that does not match the number of sheet names?

Reply:
I think you're right, we should raise an error instead, let me fix that